### PR TITLE
vue3: Add LTAR cells

### PR DESCRIPTION
**Pull request by pranavxc**
## Change Summary

-  Many to many
-  Has many
-  Belongs to
-  Children list modal
-  Link row modal

re #2909

## Change type

- [x] feat: (new feature for the user, not a new feature for build script)
- [ ] fix: (bug fix for the user, not a fix to a build script)
- [ ] docs: (changes to the documentation)
- [ ] style: (formatting, missing semi colons, etc; no production code change)
- [ ] refactor: (refactoring production code, eg. renaming a variable)
- [ ] test: (adding missing tests, refactoring tests; no production code change)
- [ ] chore: (updating grunt tasks etc; no production code change)

## Test/ Verification

Provide summary of changes.

## Additional information / screenshots (optional)

Anything for maintainers to be made aware of

### CODE REVIEW

**Comment by dstala** [comment]
- [ ] Expand link to display record information
- [ ] `Add new record` within `link to` modal
- [ ] Doesn't display primary key information along with link details in expanded form

Verified
- HM, BT, MM
- Link add/remove
- Delete link child list modal
- Unlink from child list link modal

**Review by dstala — APPROVED** [review]
Basic tests completed. WIP items listed in comments

File: packages/nc-gui-v2/composables/useLTARStore.ts [hunk 1]
```diff
@@ -0,0 +1,243 @@
...
+
+    const relatedTablePrimaryValueProp = computed(() => {
+      return (relatedTableMeta?.value?.columns?.find((c) => c.pv) || relatedTableMeta?.value?.columns?.[0])?.title
+    })
+    const primaryValueProp = computed(() => {
+      return (meta?.value?.columns?.find((c: Required<ColumnType>) => c.pv) || relatedTableMeta?.value?.columns?.[0])?.title
+    })
+
+    const loadChildrenExcludedList = async () => {
+      childrenExcludedList.value = await $api.dbTableRow.nestedChildrenExcludedList(
```
Comment (wingkwong) [inline packages/nc-gui-v2/composables/useLTARStore.ts:68]: `try .. catch` for `$api.dbTableRow.nestedChildrenExcludedList`?

File: packages/nc-gui-v2/composables/useLTARStore.ts [hunk 2]
```diff
@@ -0,0 +1,243 @@
...
+          offset: childrenExcludedListPagination.size * (childrenExcludedListPagination.page - 1),
+          where:
+            childrenExcludedListPagination.query &&
+            `(${relatedTablePrimaryValueProp.value},like,${childrenExcludedListPagination.query})`,
+        } as any,
+      )
+    }
+
+    const loadChildrenList = async () => {
+      childrenList.value = await $api.dbTableRow.nestedList(
```
Comment (wingkwong) [inline packages/nc-gui-v2/composables/useLTARStore.ts:87]: `try .. catch` for `$api.dbTableRow.nestedList`?

File: packages/nc-gui-v2/components/smartsheet/Grid.vue [hunk 3]
```diff
@@ -1,6 +1,7 @@
 <script lang="ts" setup>
-import { isVirtualCol } from 'nocodb-sdk'
+import { ColumnType, isVirtualCol } from 'nocodb-sdk'
 import {
+  Row,
```
Comment (bcakmakoglu) [inline packages/nc-gui-v2/components/smartsheet/Grid.vue:4]: Unused import?

File: packages/nc-gui-v2/components/virtual-cell/Lookup.vue [hunk 4]
```diff
@@ -11,7 +11,7 @@ provide(ReadonlyInj, true)
 const column = inject(ColumnInj) as ColumnType & { colOptions: LookupType }
 const meta = inject(MetaInj)
 const value = inject(ValueInj)
```
Comment (bcakmakoglu) [inline packages/nc-gui-v2/components/virtual-cell/Lookup.vue:13]: Can we possibly rename `ValueInj` to something like `CellValueInj`?
So it doesn't end up with `value.value`

File: packages/nc-gui-v2/components/virtual-cell/components/ListChildItems.vue [hunk 5]
```diff
@@ -1,254 +1,85 @@
-<script>
-import { RelationTypes } from 'nocodb-sdk'
-import { NOCO } from '~/lib/constants'
-import Pagination from '~/components/project/spreadsheet/components/Pagination'
+<script lang="ts" setup>
+import { useVModel } from '#imports'
+import { useLTARStoreOrThrow } from '~/composables'
```
Comment (bcakmakoglu) [inline packages/nc-gui-v2/components/virtual-cell/components/ListChildItems.vue:3]: import both from `#imports`

File: packages/nc-gui-v2/composables/useLTARStore.ts [hunk 6]
```diff
@@ -0,0 +1,257 @@
...
+import { NOCO } from '~/lib'
+import type { Row } from '~/composables'
+import { extractSdkResponseErrorMsg } from '~/utils'
+
+interface DataApiResponse {
+  list: Record<string, any>
+  pageInfo: PaginatedType
+}
+
+const [useProvideLTARStore, useLTARStore] = useInjectionState(
```
Comment (bcakmakoglu) [inline packages/nc-gui-v2/composables/useLTARStore.ts:14]: A comment what `LTAR` even means would be nice so others understand the name as well (probably Link to another Record, right?)

**Review by bcakmakoglu — CHANGES_REQUESTED** [review]

File: packages/nc-gui-v2/components/virtual-cell/Lookup.vue (hunk 4 above)
Comment (pranavxc) [reply packages/nc-gui-v2/components/virtual-cell/Lookup.vue:13]: agree i will update that :+1:

**Review by bcakmakoglu — APPROVED** [review]

**Pull request merged**